### PR TITLE
GCI responsive carousel

### PR DESCRIPTION
**Pull request by adamstafa**
I added two variables: `bp-mobile` and `bp-mobile-small` for responsive design and used it in demo.scss. Carousel now doesn't break on small screens. Headings of the demos now have a gap between lines so it doesn't look ugly when the heading is broken because of width.

### CODE REVIEW

**Comment by uttu357** [comment]
@adamstafa On the desktop site, can we have more than just one demo at a time on the screen. The mobile version looks good to me.

File: frontend/src/components/home/homeBody/HomeDemos.js [hunk 1]
```diff
@@ -86,7 +99,10 @@ class HomeDemos extends React.Component {
               {DEMOS_TITLE}
             </h1>
             {CAROUSEL
-              ? <Carousel decorators={CarouselDecorators} slidesToShow={1}>
+              ? <Carousel
+                  decorators={CarouselDecorators}
+                  slidesToShow={window.innerWidth > 1000 ? 2 : 1}
```
Comment (uttu357) [inline frontend/src/components/home/homeBody/HomeDemos.js]: Define const variables in the render method for the no. of demos on large screens and no.of demos on small screen. Save this in state of the component `window.innerWidth > 1000` and add a listener to update it. The state variable will be used here to check if window.innerWidth greater or not.

File: frontend/src/styles/partials/_home.scss [hunk 2]
```diff
@@ -195,7 +195,12 @@
             text-align: center;
             font-size: 30px;
```
Comment (uttu357) [inline frontend/src/styles/partials/_home.scss]: Use font size mixin for line height and font-size

File: frontend/src/styles/partials/_home.scss [hunk 3]
```diff
@@ -195,7 +195,12 @@
...
             font-family: $heading-font-family;
+            line-height: 32px;
             @include text-gradient($heading-color, $secondary-color);
+            @media #{$bp-mobile-small} {
+              font-size: 26px;
```
Comment (uttu357) [inline frontend/src/styles/partials/_home.scss]: Here also.

File: frontend/src/styles/partials/_home.scss [hunk 4]
```diff
@@ -221,17 +226,31 @@
...
 
+          .cv-home-demos-source {
+            padding-left: 0;
+          }
+
           .cv-home-demos-carousel-card {
             height: 210px;
             box-shadow: none;
+            @media #{$bp-mobile} {
+              height: 240px;
```
Comment (uttu357) [inline frontend/src/styles/partials/_home.scss]: Remove all pixels from your code and use rem instead.

**Comment by adamstafa** [comment]
@uttu357 Do I have to use these media queries? I found IMHO a better solution.
I get all cards with `document.getElementsByClassName(...)`, find the tallest one and set all cards the same height.
Is it OK or will it have any negative performance effect?

**Comment by uttu357** [comment]
Instead of using media queries you could use `rem`. It is set to auto adjust according to the view width of the device.

**Comment by adamstafa** [comment]
I tried the rem solution. It looks pretty good on large screens while being simple, but it starts to break on mobiles. The problem is that the description is like 5 lines high on 4-inch phone but on 5-inch one it's like 3 lines. So on small screen it's sometimes very crowded but on a bit larger screen there's too much unused space.
I would rather use setting all cards the same height as the tallest one if you don't have a problem with that. 😄

**Comment by uttu357** [comment]
@adamstafa Okay. That works too! :+1:

**Comment by adamstafa** [comment]
@uttu357 changed all the stuff you wanted. Also the demos are now more modular and easier to change. Height of the cards is now adaptive so they won't break. The `equalizeCards` method is a bit cancer but I couldn't find a better solution. Hopefully you won't ever have to change it 😀 

In method `updateCarousel` you can change count of slides to show. In `makeDescription` you can set limit to whatever you want, the cards won't break.

IDK what to do with the dots under carousel tho. Currently there are displayed `slidesCount - (slidesToShow - 1)` dots  (one for each "state") I could display `slidesCount` dots and highlight `slidesToShow` dots. Just tell me what you like more and I'll change it 😄

**Comment by uttu357** [comment]
LGTM! Merging.. :+1:

**Pull request merged**